Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -2916,4 +2916,22 @@ class HTTPClientTests: XCTestCase {
             XCTAssertEqual(error as? ExpectedError, .expected)
         }
     }
+    
+    func testRequestSpecificTLS() throws {
```

Comment:
could we make a second request in this test and check the `RequestInfo` that the test server should return.

The `RequestInfo` has the following fields:

```swift
internal struct RequestInfo: Codable {
    var data: String
    var requestNumber: Int
    var connectionNumber: Int
}
```

so you should be able to write

```swift
        guard let response = maybeResponse, let body = response.body else {
            XCTFail("request failed")
            return
        }
        let connectionNumber = try? JSONDecoder().decode(RequestInfo.self, from: body).connectionNumber
```

and then (in a few test cases) we can validate that:

- two subsequent requests with _different_ `TLSConfiguration`s get different `connectionNumber`s
- two subsequent requests with the _same_ `TLSConfiguration` get the same `connectionNumber` for both requests.

Reply:
Sure! Did not spot that the we could test it that way, good to know! I added the assertions to the same test, lmk if you meant something else!

Reply:
thank you!